src/TriggerApplication.cpp

```diff
@@ -0,0 +1,509 @@
+{
+  auto ntServiceObj = ntDesc->get_associated_service()->config_object();
+  std::string ntUid(idname);
+  oksdbinterfaces::ConfigObject ntObj;
+  confdb->create(dbfile, "NetworkConnection", ntUid, ntObj);
+  ntObj.set_by_val<std::string>("connection_type", ntDesc->get_connection_type());
+  ntObj.set_obj("associated_service", &ntServiceObj);
+
+  return ntObj;
+}
+
+std::vector<const coredal::DaqModule*> 
+TriggerApplication::generate_modules(oksdbinterfaces::Configuration* confdb,
+                                     const std::string& dbfile,
+                                     const coredal::Session* session) const 
+{
+  std::vector<const coredal::DaqModule*> modules;
+
+  // Trigger Interrupt network descriptor
+  const NetworkConnectionDescriptor* tiMLTNetDesc = nullptr;
+  // TriggerDecisions network descriptor
+  const NetworkConnectionDescriptor* tdMLTNetDesc = nullptr;
+  // Timing trigger network descriptor
+  const NetworkConnectionDescriptor* tmgTrgNetDesc = nullptr;
+
+  // TC queue descriptor for link to the MLT
+  //const QueueDescriptor* tcMLTQueueDesc;
+  //const QueueDescriptor* rtcmQueueDesc;
+  const QueueDescriptor* tcQueueDesc;
+
+  /**************************************************************
+   * Get all the queue connection descriptions
+   **************************************************************/
+  for (auto rule : get_queue_rules()) {
+    std::string destination_class = rule->get_destination_class();
+    std::string data_type = rule->get_descriptor()->get_data_type();
+    if((destination_class == "ModuleLevelTrigger") && (data_type == "TriggerCandidate"))
+      tcQueueDesc = rule->get_descriptor();
+  }
+  if(!tcQueueDesc)
+    throw (BadConf(ERS_HERE, "No description for the queue of TCs going tinto the MLT given"));
+
+  /**************************************************************
+   * Get all the network connections
+   **************************************************************/
+  size_t nReadoutLinks = 0;
+  for(auto rule : get_network_rules()){
+    std::string endpoint_class = rule->get_endpoint_class();
+    std::string data_type = rule->get_descriptor()->get_data_type();
+
+    // Network connections for the MLT
+    if(data_type == "TriggerInhibit")
+      tiMLTNetDesc = rule->get_descriptor();
+    if(data_type == "TriggerDecision")
+      tdMLTNetDesc = rule->get_descriptor();
+    if(data_type == "HSIEvent")
+      tmgTrgNetDesc = rule->get_descriptor();
+
+    std::cout << "Endpoint class: " << endpoint_class << " data_type: " << data_type << std::endl;
+  }
+
+  if(!tdMLTNetDesc)
+    throw (BadConf(ERS_HERE, "No MLT network connection for the output TriggerDecision given"));
+
+  /**************************************************************
+   * Get the MLT
+   **************************************************************/
+  auto mlt_conf = get_mlt_conf();
+  // Don't build the rest if we have no MLT
+  if(!mlt_conf)
+    return modules;
```

Comment:
If you don't have an MLT config you return an empty module list? Isn't this an invalid configuration that should throw a BadConf Issue?